ActiveLabel/ActiveLabel.swift

```diff
@@ -139,22 +148,141 @@ typealias ElementTuple = (range: NSRange, element: ActiveElement, type: ActiveTy
         didSet { textContainer.lineBreakMode = lineBreakMode }
     }
 
+    private func setupLongPressGesture() {
+        self.longPressGesture.minimumPressDuration = 0.5
+        self.longPressGesture.addTarget(self, action: #selector(self.longPress(_:)))
+        self.addGestureRecognizer(self.longPressGesture)
+    }
+    
+    @objc private func longPress(_ sender: UILongPressGestureRecognizer) {
+        let location = sender.location(in: sender.view)
+        if let element = element(at: location) {
+            switch element.element {
+            case .mention(let userHandle):
+                guard let url = URL(string: "@" + userHandle) else {
+                    return
+                }
+                self.processLink(url, touchPoint: location, sender: sender)
+            case .hashtag(let hashtag):
+                guard let url = URL(string: "#" + hashtag) else {
+                    return
+                }
+                self.processLink(url, touchPoint: location, sender: sender)
+            case .url(let originalURL, _):
+                guard let url = URL(string: originalURL) else {
+                    return
+                }
+                self.processLink(url, touchPoint: location, sender: sender)
+            case .custom(_):
+                break
+            }
+        }
+    }
+
+    private func processLink(_ url: URL!, touchPoint: CGPoint, sender: UILongPressGestureRecognizer) {
+        self.delegate?.didLongPressWithURL(url, touchPoint: touchPoint)
+        guard self.copyLinksActive,
+            let rect = self.selectedLinkRectangle(link: url.description, touchPoint: touchPoint),
+            self.lastCopyMenuRect != rect else {
+            return
+        }
+        self.lastCopyMenuRect = rect
+        self.showCopyMenu(rect: rect, sender: sender)
+        self.copyLink = url.description
+    }
+    
+    private func selectedLinkRectangle(link: String, touchPoint: CGPoint) -> CGRect? {
+        guard let text = self.text else {
+            return nil
+        }
+        let linkRanges = self.processLinkRanges(text: text, link: link)
+        for linkRange in linkRanges {
+            let characterRange = NSRange(linkRange, in: text)
+            let boundRect = processBoundRect(forCharacterRange: characterRange)
+            let correctBoundRect = CGRect(x: boundRect.minX, y: boundRect.minY + self.heightCorrection, width: boundRect.width, height: boundRect.height)
+            guard !correctBoundRect.contains(touchPoint) else {
+                self.highlightLink(range: characterRange)
+                return correctBoundRect
+            }
+        }
+        return nil
+    }
+    
+    private func processLinkRanges(text: String, link: String) -> [Range<String.Index>] {
```

Comment:
processLinkRanges means that this function will do something with ranges but in fact it seeks ranges by text and link. I suggest to rename it to `linkRanges()`